url.bs

```diff
@@ -255,6 +255,92 @@ point <a for=url>URLs</a> from <var>A</var> can come from untrusted sources.
 <a for=host>host</a> serves as a network address, but it is sometimes (ab)used as opaque
 identifier in <a for=url>URLs</a> where a network address is not necessary.
 
+<p>A <a>host</a>'s <dfn export for=host id=concept-host-public-suffix>public suffix</dfn> is
+the portion of a <a>host</a> which is controlled by a registrar, public or otherwise.
+To obtain a <var>host</var>'s <a>public suffix</a>, run these steps:
+
+1.  If <var>host</var> is an <a for=host>IPv4 address</a> or a <a for=host>IPv6 address</a>
+    return the empty string.
+
+2.  Let <var>host labels</var> be the result of <a lt="strictly split">strictly splitting</a>
+    <var>host</var> on the U+002E FULL STOP ('<code>.</code>') character.
+
+3.  Apply the <a>domain to ASCII</a> algorithm to each token in <var>host labels</var>.
+
+4.  Let <var>host</var> be the result of concatenating the items in <var>host labels</var>, in
+    the same order, separated by the U+002E FULL STOP ('<code>.</code>') character.
+
+5.  While <var>host</var> is not the empty string:
+
+    1.  If <var>host</var> matches the end of a suffix in the Public Suffix List,
+        return <var>host</var>. [PSL]
```

Comment:
This doesn't seem correct.

https://publicsuffix.org/list/ is the canonical description of the algorithm, which explains the list format and a formal algorithm and terminology definition. See "Algorithm"